adapter/src/utils/useLocale.js

```diff
@@ -23,22 +52,41 @@ const setGlobalLocale = (locale) => {
     }
     moment.locale(locale)
 
-    const simplifiedLocale = simplifyLocale(locale)
-    i18n.changeLanguage(simplifiedLocale)
+    const resolvedLocale = validateLocaleByBundle(locale)
+    i18n.changeLanguage(resolvedLocale)
+
+    console.log('🗺 Global d2-i18n locale initialized:', resolvedLocale)
+}
+
+// Sets the global direction based on the app's configured direction
+// (which should be done to affect modals, alerts, and other portal elements),
+// then returns the locale's direction for use on the header bar
+const handleDirection = ({ locale, configDirection }) => {
+    // for i18n.dir, need JS-formatted locale
```

Comment:
Also for all of this, maybe we can be more defensive and wrap it in a try/catch and fallback to 'ltr' - there are few external inputs here (the Jav locale) that might go wrong so it'd be good to make sure it doesn't crash.

Reply:
It's possible `i18n.dir` is already quite robust in that way -- you can pass it any string (or an array even, lol), and it will default to `'ltr'`

Was there something else here you think is fragile?

Reply:
I guess I had `locale` being undefined in mind .. I thought it might be on first-render or something, and it's better to be safe. But maybe a try/catch around the whole useLocale effect is even safer (the stuff with splitting locale strings could throw as well in a couple of places)

Reply:
In my recent changes, I approached error handling in a bit more specific way at each step with useful fall-backs -- my idea is that if one step doesn't work, the other steps can still succeed and provide useful localization. Not many things in here can actually throw, but those things are wrapped in try/catch